File: Java/DFS.java
import java.util.*;

public class DFS {
    private Map<String, List<String>> graph = new HashMap<>();

    public void addEdge(String source, String destination) {
        graph.putIfAbsent(source, new ArrayList<>());
        graph.putIfAbsent(destination, new ArrayList<>());
        graph.get(source).add(destination);
        graph.get(destination).add(source);
    }

    public void dfs(String start) {
        Set<String> visited = new HashSet<>();
        dfsRecursive(start, visited);
    }

    private void dfsRecursive(String node, Set<String> visited) {
        visited.add(node);
        System.out.print(node + " ");
        for (String neighbor : graph.get(node)) {
            if (!visited.contains(neighbor)) {
                dfsRecursive(neighbor, visited);
            }
        }
    }

    public static void main(String[] args) {
        DFS dfsExample = new DFS();
        dfsExample.addEdge("A", "B");
        dfsExample.addEdge("A", "C");
        dfsExample.addEdge("B", "D");
        dfsExample.addEdge("B", "E");
        dfsExample.addEdge("C", "F");
        dfsExample.addEdge("E", "F");

        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the start node: ");
        String startNode = scanner.nextLine();

        System.out.print("DFS traversal: ");
        dfsExample.dfs(startNode);
        System.out.println();
    }
}

File: C#/dfs.cs
using System;
using System.Collections.Generic;

class Graph
{
    private int V;
    private List<int>[] adj;

    public Graph(int v)
    {
        V = v;
        adj = new List<int>[v];
        for (int i = 0; i < v; ++i)
            adj[i] = new List<int>();
    }

    public void AddEdge(int v, int w)
    {
        adj[v].Add(w);
    }

    public void DFSUtil(int v, bool[] visited)
    {
        visited[v] = true;
        Console.Write(v + " ");

        foreach (int n in adj[v])
        {
            if (!visited[n])
                DFSUtil(n, visited);
        }
    }

    public void DFS(int v)
    {
        bool[] visited = new bool[V];

        DFSUtil(v, visited);
    }

    public static void Main()
    {
        Graph g = new Graph(4);

        g.AddEdge(0, 1);
        g.AddEdge(0, 2);
        g.AddEdge(1, 2);
        g.AddEdge(2, 0);
        g.AddEdge(2, 3);
        g.AddEdge(3, 3);

        Console.Write("DFS traversal starting from vertex 2: ");
        g.DFS(2);
        Console.WriteLine();
    }
}
